stream.go

```diff
@@ -459,17 +464,34 @@ func (cs *clientStream) commitAttempt() {
 // shouldRetry returns nil if the RPC should be retried; otherwise it returns
 // the error that should be returned by the operation.
 func (cs *clientStream) shouldRetry(err error) error {
+	unprocessed := false
+	if cs.attempt.s == nil {
+		pioErr, ok := err.(transport.PerformedIOError)
+		if ok {
+			// Unwrap error.
+			err = toRPCErr(pioErr.Err)
+		} else {
+			unprocessed = true
+		}
+		if !ok && !cs.callInfo.failFast {
```

Comment:
In what situation would we want to fail non-wait-for-ready RPC when there's no IO?

Reply:
Per the spec, we are only allowed 1 transparent retry for non-WFR RPCs.  The idea being that they should be making progress at all times.